### DISCUSSION TITLE
libp2p-rs compares to iroh-rs

### DISCUSSION THREAD

**Discussion by ghost** [libp2p/rust-libp2p#5247] [Q&A]
I work with both libp2p-rs and [iroh-rs](https://github.com/n0-computer/iroh) now and then. I am so into with libp2p but recently I have worked a lot with iroh and I see that it's more faster. I have my own thoughts about both platform but I want to see from the libp2p's main contributors' views either. Thank you!

**Comment by dariusc93** [comment]
I've played with iroh (both when they used libp2p (in their now archived project called beetle) and their new code) and I do feel both have their pros and cons but also is aimed at different use-cases.

**Comment by jxs** [comment]
Hi @zrus 

Iroh and libp2p operate on different layers, as [Irohs README.md states](https://github.com/n0-computer/iroh/blob/main/README.md?plain=1#L41) "Iroh is a protocol for syncing & moving bytes" libp2p OTOH libp2p is a modular system of transports and spec'd protocols that enable development of other different types of applications.

Iroh's first implementation [used libp2p](https://github.com/n0-computer/beetle/blob/main/Cargo.toml#L112). After the [new direction ](https://iroh.computer/blog/a-new-direction-for-iroh) AFAIK Iroh opted to focus [on QUIC](https://iroh.computer/docs/layers/connections#connections) for the transport discarding libp2p's [multistream-select](https://github.com/libp2p/specs/blob/master/connections/README.md#upgrading-connections)
